transfer_learning.ipynb

```diff
@@ -372,31 +350,31 @@
    },
    "outputs": [],
    "source": [
+    "@torch.no_grad()\n",
     "def visualize_model(model, num_images=6):\n",
     "    was_training = model.training\n",
     "    model.eval()\n",
     "    images_so_far = 0\n",
     "    fig = plt.figure()\n",
     "\n",
-    "    with torch.no_grad():\n",
-    "        for i, (inputs, labels) in enumerate(dataloaders['val']):\n",
-    "            inputs = inputs.to(device)\n",
-    "            labels = labels.to(device)\n",
-    "\n",
-    "            outputs = model(inputs)\n",
-    "            _, preds = torch.max(outputs, 1)\n",
-    "\n",
-    "            for j in range(inputs.size()[0]):\n",
-    "                images_so_far += 1\n",
-    "                ax = plt.subplot(num_images//2, 2, images_so_far)\n",
-    "                ax.axis('off')\n",
-    "                ax.set_title('predicted: {}'.format(class_names[preds[j]]))\n",
-    "                imshow(inputs.cpu().data[j])\n",
-    "\n",
-    "                if images_so_far == num_images:\n",
-    "                    model.train(mode=was_training)\n",
-    "                    return\n",
-    "        model.train(mode=was_training)"
+    "    for i, (inputs, labels) in enumerate(dataloaders['val']):\n",
+    "        inputs = inputs.to(device)\n",
+    "        labels = labels.to(device)\n",
+    "\n",
+    "        outputs = model(inputs)\n",
+    "        _, preds = torch.max(outputs, 1)\n",
+    "\n",
+    "        for j in range(inputs.size(0)):\n",
+    "            images_so_far += 1\n",
+    "            ax = plt.subplot(num_images // 2, 2, images_so_far)\n",
+    "            ax.axis('off')\n",
+    "            ax.set_title('predicted: {}'.format(class_names[preds[j]]))\n",
+    "            imshow(inputs[j])\n",
```

Comment:
Everything in this block is the same, except this line, where I changed `inputs.cpu().data[j]` to `inputs[j]`. This works because in `imshow` I added `.detach().cpu()` to make it slightly more robust